### PR TITLE
feat(errorutil): add MeshKit PR validation and pre-commit hook

### PR DESCRIPTION
**Pull request by SuryanshGarg04**
This PR represents the final phase of the errorutil validation effort, following [[#1080](https://github.com/meshery/meshkit/pull/1080)](https://github.com/meshery/meshkit/pull/1080), [[#1105](https://github.com/meshery/meshkit/pull/1105)](https://github.com/meshery/meshkit/pull/1105), and [[Meshery #21009](https://github.com/meshery/meshery/pull/21009)](https://github.com/meshery/meshery/pull/21009), by adding local pre-commit and PR-level CI validation natively for MeshKit.

Previously, MeshKit did not have developer-side pre-commit validation or PR-level error-code validation. This PR closes that gap while reusing the allocation-aware validation introduced in #1105.

### Key Changes

* **CI Enforcement**: `error-codes-updater.yaml` gains a `pull_request` validation path that runs `errorutil check` against the PR's merge-base. The existing push-triggered allocation/update behavior remains isolated to push events.
* **Pre-commit Hook**: Added an opt-in local Git hook at `.githooks/pre-commit`, installed via `make setup-hooks`. The hook is read-only and validates the exact staged index state against `HEAD` using the same `errorutil check` command as CI. It supports both `replace_me` and valid locally allocated integer codes, while rejecting new integer codes that are inconsistent with the recorded allocation state. It can be bypassed with `git commit --no-verify`; CI remains the authoritative enforcement layer.
* **Documentation**: Extended `docs/agent-instructions/errors.md` with the hook workflow and CI behavior.
* **Hook Test Cases**: Added an 11-case suite covering Git-level edge cases such as staged vs. unstaged changes, partial staging, untracked working-tree changes, unrelated files, and baseline/index reconstruction.

### Relationship to #1105

This PR does not introduce another validation implementation. It consumes the allocation-aware `errorutil check` behavior from #1105 and uses that same validator for both MeshKit pre-commit and PR CI enforcement.

### Scope

This PR does not introduce:

* a new error-code allocation mechanism
* automatic allocation during validation
* automatic repair of existing integer collisions
* Merge Queue or global allocation locking
* cleanup of historical duplicate-code debt

## Summary by CodeRabbit

* **New Features**
  * Added optional local pre-commit validation for staged error-code changes, installable with `make setup-hooks`.
  * Added pull request checks against the appropriate baseline to validate error-code updates.
  * Enhanced detection of duplicate, invalid, or incorrectly allocated error codes, including counter and allocation-window issues.
  * Improved validation for staged, renamed, deleted, and locally allocated error-code changes.

* **Documentation**
  * Documented hook installation, validation behavior, bypass options, and pull request checks.

### CODE REVIEW

File: .githooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# Validates the STAGED (index) state against HEAD using the same
+# `errorutil check` that runs in CI (.github/workflows/error-codes-updater.yaml).
+# Does not allocate codes, does not modify any file, and is not the security
+# boundary - `git commit --no-verify` bypasses this; CI remains authoritative.
+set -eu
+
+PAT_ERR='*.go'
+PAT_INFO='helpers/component_info.json'
+
+staged=$(git diff --cached --name-only --diff-filter=ACMR -- $PAT_ERR $PAT_INFO | grep -v '_test\.go$' || true)
```
[the remark this answers is not in the document]
Comment (SuryanshGarg04) [reply .githooks/pre-commit]: addressed in `9c7876d`
